text/XXXX-quality-check-workflow.md

```diff
@@ -0,0 +1,546 @@
+| ------------------------------- | ----------- | ------------------------------------------------------------------------------ | ---------------------------------------------------- | ---------------------------------------------------- | ---------------------------------------------------- |
+| `ngsderive instrument`          | [ngsderive] | The predicted sequencing instrument for this file.                             | ![yes](https://img.shields.io/badge/yes-brightgreen) | ![yes](https://img.shields.io/badge/yes-brightgreen) | ![yes](https://img.shields.io/badge/yes-brightgreen) |
+| `ngsderive readlen`             | [ngsderive] | The predicted read length for the file.                                        | ![yes](https://img.shields.io/badge/yes-brightgreen) | ![yes](https://img.shields.io/badge/yes-brightgreen) | ![yes](https://img.shields.io/badge/yes-brightgreen) |
+| `ngsderive encoding`            | [ngsderive] | The predicted FASTQ quality score encoding scheme for the file.                | ![yes](https://img.shields.io/badge/yes-brightgreen) | ![yes](https://img.shields.io/badge/yes-brightgreen) | ![yes](https://img.shields.io/badge/yes-brightgreen) |
+| `ngsderive strandedness`        | [ngsderive] | The predicted strandedness of the RNA-Seq protocol for the file.               | ![no](https://img.shields.io/badge/no-red)           | ![no](https://img.shields.io/badge/no-red)           | ![yes](https://img.shields.io/badge/yes-brightgreen) |
+| `ngsderive junction-annotation` | [ngsderive] | The predicted breakdown of novel, partially novel, and known splice junctions. | ![no](https://img.shields.io/badge/no-red)           | ![no](https://img.shields.io/badge/no-red)           | ![yes](https://img.shields.io/badge/yes-brightgreen) |
+
+`ngsderive` is a tool written by the St. Jude Cloud team to provide forensic
+analysis techniques for evaluating next-generation sequencing data. The
+following subcommands are particularly useful in the context of these QC
+reports:
+
+- **`ngsderive instrument`.** Gives an estimate of which sequencing instrument
+  was used for a particular file. You should check to ensure the result you
+  receive from this subcommand matches your expectations of the data.
+- **`ngsderive readlen`.** Gives an estimate of the pre-trimmed read length. For
+  files with significant adapter trimming, this may not provide conclusive
+  results (giving a result of `-1`). You should check to ensure the result you
+  receive from this subcommand matches your expectations of the data.
+- **`ngsderive encoding`.** Guesses which [FASTQ
+  encoding](https://en.wikipedia.org/wiki/FASTQ_format#Encoding) was used for
+  the original FASTQ file. For most modern data, this should be `Sanger/Illumina
+  1.8`.
```

Comment:
Agree this blurb needs some work. Not sure if your question was rhetorical or sincere, but it's a sanity check (which I've never seen fail out of thousands of samples). These two sentences from the `ngsderive` [wiki](https://stjudecloud.github.io/ngsderive/subcommands/encoding/) succinctly explain the point:
> The BAM specification calls for PHRED+33, however this is also the most permissive encoding. It is possible for a stricter encoding to be mis-translated as PHRED+33 and appear to be of higher quality than it is in truth.

From reading the STAR manual, I know it's possible to create a BAM with an alternative encoding. I could never think of a case where you'd intentionally want to do that, but it's possible in theory and this command checks for it.

I'm also not really a fan of the word `Guesses` here. It's a more definitive algorithm than that phrasing implies. There are only 2 possibilities if an encoding other than `PHRED+33` is returned:
1) an alternative encoding was used in the FASTQ(s) and it was mistranslated during alignment
2) every base checked by `ngsderive` just happened to be of extremely high quality, and there wasn't a single low-to-moderate quality base call in the entire sample size

Obviously number 2 is statistically improbable (I supposed unless there was some bizarro base-call filtering done on your BAM), so if we ever saw a non-`PHRED+33` encoding in production, it's a very safe assumption that the encoding was mis-translated.

Reply:
It was intended sincerely, however, upon rereading, I wasn't clear in my question. I was specifically referring to the end user. I don't think this metric gives them any useful information. So I think this goes back to a question about how this RFC should be written. My thought is that for each metric it should address 1. what the metric measures, 2. what it tells us, 3. how we use the metric to guide our choices (cutoffs, inclusion/exclusion, etc.), and 4. what consideration an end-user should give to the metric.

I initially missed `guesses`, but I absolutely agree with @a-frantz. We need to rethink how we describe it.

Reply:
Just did a reread of this original text, and realized it's not even correct about what `encoding` does. It does NOT:
> Guess which FASTQ encoding was used for the original FASTQ file

That information could only be derived from the BAM header, specifically looking at the command which aligned the FASTQ files. However, I don't see in what context that would ever be useful information.

`ngsderive encoding` _can_ be used on FASTQ files, which are allowed to have a variety of quality encodings, in order to know how to properly set the parameters to your aligner of choice. But this is an aside to the RFC we're discussing, since we only run `encoding` on BAM files (which _should_ all return `Sanger/Illumina 1.8` if following the BAM spec).

How this section gets re-written depends on who the audience of the document is.
* If this is meant to document "how the SJ Cloud team runs QC", this section would be written to explain the sanity-check nature of this result. i.e. it should always be `Sanger/Illumina 1.8`, and any other result indicates a problem that must be triaged.
* If this is meant to document "how do end-users use our QC data?", the answer is they ignore this result. It's completely unimportant.
    * speculation on my part: but I can't imagine a scenario where we as a team would allow a non PHRED+33 encoded BAM to be hosted on the cloud. We would re-align the file and set the encoding properly. The repercussion of that is that *every* BAM we host on the cloud is PHRED+33 encoded, and there's never a reason for a user to check this particular metric (it's going to be the same for every file).